ntoskrnl/include/internal/ntoskrnl.h

```diff
@@ -103,6 +103,12 @@
 
 #endif
 
+#ifdef __SANITIZE_UB__
+#define NO_SANITIZE __attribute__ ((no_sanitize("undefined")))
```

Comment:
I'd prefer to have it in crtdefs.h

Reply:
Ahemmm... until we see whether there are other functions in our code base that may need the sanitizer to not be applied on them, it might be better to keep that only where it's needed (so far just the kernel), and not pollute our CRT with unrelated stuff. And even if you would want to move that into some global header, a specific separate one might be better.

Reply:
I have some :smile:

Reply:
Dunno... I see sanitizer stuff as something separate from compiler 🤷

Reply:
Well, it is the compiler instrumenting generated binaries in order to make the relevant checks. It can't be closer to compiler than that :wink:

Reply:
@HBelusca check this article for example - explains how Address Sanitizer works, for example https://lwn.net/Articles/612153/